File: Selenium/Activities/Activity8.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as fireFoxService
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
service=fireFoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service)as driver:
    driver.get("https://training-support.net/selenium/dynamic-controls")
    print("Page Title:- " + driver.title)

    toggleCheckbox=driver.find_element(By.ID,"toggleCheckbox")
    toggleCheckbox.click()
    wait=WebDriverWait(driver,20)

    wait.until(EC.presence_of_element_located([By.XPATH,"//button[text()='Add Checkbox']"]))
    print(toggleCheckbox.is_displayed())

    driver.find_element(By.XPATH,"//button[text()='Add Checkbox']").click()
    wait.until(EC.presence_of_element_located([By.XPATH,"//button[text()='Remove Checkbox']"]))
    print(driver.find_element(By.XPATH,"//button[text()='Remove Checkbox']").is_displayed)
    driver.close()

    
    

File: Selenium/Activities/Activity8.java
package Activity;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity8 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver=new FirefoxDriver();
        driver.get(" https://training-support.net/selenium/dynamic-controls");
        System.out.println("Page title :- " + driver.getTitle());

        WebElement toggleCheckbox = driver.findElement(By.id("toggleCheckbox"));
        toggleCheckbox.click();
        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//button[text()='Add Checkbox']")));
        System.out.println(toggleCheckbox.isDisplayed());
        driver.findElement(By.xpath("//button[text()='Add Checkbox']")).click();
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//button[text()='Remove Checkbox']")));
        System.out.println(driver.findElement(By.xpath("//button[text()='Remove Checkbox']")).isDisplayed());
        driver.close();

    }
}
